tools/github-event-processor/Azure.Sdk.Tools.GitHubEventProcessor/GitHubEventClient.cs

```diff
@@ -0,0 +1,806 @@
+                                                       int daysSinceLastUpdate = 0,
+                                                       List<IssueIsQualifier> issueIsQualifiers = null,
+                                                       List<string> labelsToInclude = null,
+                                                       List<string> labelsToExclude = null)
+        {
+            var request = new SearchIssuesRequest();
+
+            // The repo owner 
+            request.Repos.Add(repoOwner, repoName);
+
+            // Can only search for opened or closed
+            request.State = itemState;
+            if (null != issueIsQualifiers)
+            {
+                request.Is = issueIsQualifiers;
+            }
+
+            // restrict the search to issues (IssueTypeQualifier.Issue)
+            // or pull requests (IssueTypeQualifier.PullRequest)
+            request.Type = issueType;
+
+            if (daysSinceLastUpdate > 0)
+            {
+                // Octokit's DateRange wants a DateTimeOffset as other constructors are depricated
+                // AddDays of 0-days to effectively subtract them.
+                DateTime daysAgo = DateTime.UtcNow.AddDays(0 - daysSinceLastUpdate);
+                DateTimeOffset daysAgoOffset = new DateTimeOffset(daysAgo);
+                request.Updated = new DateRange(daysAgoOffset, SearchQualifierOperator.LessThan);
+            }
+
+            if (null != labelsToInclude)
+            {
+                request.Labels = labelsToInclude;
+            }
+
+            if (null != labelsToExclude)
+            {
+                // This is how things would get exluded. Anything that needs to be an exclusion
+                // for the query needs added to a SearchIssuesRequestExclusions and then
+                // the Exclusions on the request needs to be set to that.
+                var exclusions = new SearchIssuesRequestExclusions();
+                exclusions.Labels = labelsToExclude;
+                request.Exclusions = exclusions;
+            }
+            return request;
+        }
+
+        /// <summary>
+        /// Execute the query for a given SearchIssuesRequest. It was necessary to break up the SearchIssuesRequest
+        /// and the query due to pagination. The SearchIssuesResult will only contain to up the first 100 results.
+        /// Subsequent results need to be requeried with the SearchIssuesRequest.Page incremented to get the next 100
+        /// results and so on.
+        /// </summary>
+        /// <param name="searchIssuesRequest">SearchIssuesRequest objected which contains the search criteria.</param>
+        /// <returns>OctoKit.SearchIssuesResult</returns>
+        public virtual async Task<SearchIssuesResult> QueryIssues(SearchIssuesRequest searchIssuesRequest)
+        {
+            int maxTries = 5;
+            // 61 seconds
+            int sleepDuration = 61000;
```

Comment:
@weshaggard instead of 90 I went for 61 which is the bare minimum that ensures we'll wait for a limit reset if that's the cause of the secondary rate limit.